qiita_ware/search.py

```diff
@@ -0,0 +1,103 @@
+from collections import Counter, defaultdict
+
+from future.utils import viewvalues
+
+from qiita_db.study import Study
+from qiita_db.data import ProcessedData
+from qiita_db.search import QiitaStudySearch
+
+
+def count_metadata(results, meta_cols):
+    """Counts the metadata found in a search, and returns these counts
+
+    Parameters
+    ----------
+    results : dict of lists of list
+        results in the format returned by the qiita_db search obj
+    meta_cols : list
+        metadata column names searched for, as returned by qiita_db search obj
+
+    Returns
+    -------
+    fullcount : dict of dicts
+        counts for each found metadata value over all studies, in the format
+        {meta_col1: {value1: count, value2: count, ...}, ...}
+    studycount : dict of dict of dicts
+        counts for each found metadata value for each study, in the format
+        {study_id: {meta_col1: {value1: count, value2: count, ...}, ...}, ...}
+    """
+    fullcount = {}
+    # zip all samples so that each metadata column found is its own list
+    meta_vals = zip(*[samples[sample] for samples in viewvalues(results)
+                    for sample in range(len(samples))])
```

Comment:
This is somewhat hard to understand, and in the past we have shied away from using nested for loops in list comprehensions. I would break this up, but it would be good to get others' opinions